src/code_e79b0_len_1920.c

```diff
@@ -64,41 +86,194 @@ s32 does_script_exist(s32 id) {
+f32 get_global_timespace(void) {
+    //return gGlobalTimeSpace;
+    __asm__("LUI $at,0x802e;"
+        "LWC1 $f0,-0x6358($at);"
+    );
+}
+*/
 
-INCLUDE_ASM(code_e79b0_len_1920, set_script_group);
+void set_script_group(script_context* script, s8 groupFlags) {
+    script->groupFlags = groupFlags;
+}
 
 INCLUDE_ASM(code_e79b0_len_1920, bind_trigger);
 
 INCLUDE_ASM(code_e79b0_len_1920, bind_trigger_1);
 
-INCLUDE_ASM(code_e79b0_len_1920, suspend_group_script);
+void suspend_group_script(script_context* script, s32 groupFlags) {
+    int i;
+    script_context* scriptContextPtr;
+    script_context* childScript = script->childScript;
 
-INCLUDE_ASM(code_e79b0_len_1920, resume_group_script);
+    if (childScript != NULL) {
+        suspend_group_script(childScript, groupFlags);
+    }
 
-INCLUDE_ASM(code_e79b0_len_1920, suspend_all_script);
+    for(i=0; i < ARRAY_COUNT(gCurrentScriptListPtr); i++) {
+        scriptContextPtr = (*gCurrentScriptListPtr)[i];
+        if (scriptContextPtr != NULL && scriptContextPtr->parentScript == script) {
+            suspend_group_script(scriptContextPtr, groupFlags);
+        }
+    }
 
-INCLUDE_ASM(code_e79b0_len_1920, resume_all_script);
+    if ((script->groupFlags & groupFlags) != 0) {
+        script->state |= 0x2;
+    }
+}
 
-INCLUDE_ASM(code_e79b0_len_1920, suspend_group_script_index);
+void resume_group_script(script_context* script, s32 groupFlags) {
+    int i;
+    script_context* scriptContextPtr;
+    script_context* childScript = script->childScript;
 
-INCLUDE_ASM(code_e79b0_len_1920, resume_group_script_index);
+    if (childScript != NULL) {
+        resume_group_script(childScript, groupFlags);
+    }
 
-INCLUDE_ASM(code_e79b0_len_1920, suspend_all_group);
+    for(i=0; i < ARRAY_COUNT(gCurrentScriptListPtr); i++) {
+        scriptContextPtr = (*gCurrentScriptListPtr)[i];
+        if (scriptContextPtr != NULL && scriptContextPtr->parentScript == script) {
+            suspend_group_script(scriptContextPtr, groupFlags);
+        }
+    }
 
-INCLUDE_ASM(code_e79b0_len_1920, resume_all_group);
+    if ((script->groupFlags & groupFlags) != 0) {
+        script->state &= 0xFD;
+    }
+}
 
-INCLUDE_ASM(code_e79b0_len_1920, suspend_group_others);
+s32 suspend_all_script(s32 id) {
+    s32 i;
+    script_context* scriptContextPtr;
+
+    for (i=0; i < ARRAY_COUNT(gCurrentScriptListPtr); i++) {
+        scriptContextPtr = (*gCurrentScriptListPtr)[i];
+        if (scriptContextPtr != NULL && scriptContextPtr->uniqueID == id) {
+            suspend_group_script(scriptContextPtr, 0xEF);
+        }
+    }
+}
 
-INCLUDE_ASM(code_e79b0_len_1920, resume_group_others);
+s32 resume_all_script(s32 id) {
+    s32 i;
+    script_context* scriptContextPtr;
+
+    for (i=0; i < ARRAY_COUNT(gCurrentScriptListPtr); i++) {
+        scriptContextPtr = (*gCurrentScriptListPtr)[i];
+        if (scriptContextPtr != NULL && scriptContextPtr->uniqueID == id) {
```

Comment:
I'll stop commenting it but yeah all these should be id == blah imo

Reply:
This one actually works

Just kidding. It also has to be that way. Sorry Nintendo hates your preferred styling! :P